File: final_project-master/src/main/java/Openconnection/example/demo/beans/Customer.java
package Openconnection.example.demo.beans;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "customers")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int customerID;

    @NotNull
    @Size(min = 3, max = 20)
    @Column(nullable = false, length = 20)
    private String firstName;

    @Size(min = 3, max = 20)
    @Column(nullable = false, length = 40)
    private String lastName;

    @NotNull
    @Size(min = 5, max = 50)
    @Column(nullable = false, length = 40)
    private String email;

    @NotNull
    @Size(min = 6, max = 20, message = "Password length must be between 6 and 20 characters")
    @Column(nullable = false, length = 40)
    private String password;
    @JsonIgnore
    @ManyToMany(cascade = CascadeType.REMOVE)
    @JoinTable(name = "customers_coupons", joinColumns = @JoinColumn(name = "customerId"),

            inverseJoinColumns = @JoinColumn(name = "couponId"))

    @Singular

    private List<Coupon> coupons;


}

File: CouponsWebsite/src/Components/Models/Customer.ts
export class Customer {
    customerID: number;
    firstName: string;
    lastName: string;
    email: string;
    password: string;
    coupons: Coupon[];

    constructor(customerID: number, firstName: string, lastName: string, email: string, password: string, coupons: Coupon[]) {
        this.customerID = customerID;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.password = password;
        this.coupons = coupons;
    }
}

export class Coupon {
}
